controller/controller_manager.go

```diff
@@ -188,6 +185,60 @@ func GetGuaranteedResourceRequirement(ds *datastore.DataStore) (*corev1.Resource
 	}, nil
 }
 
+func GetInstanceManagerCPURequirement(ds *datastore.DataStore, imName string) (*corev1.ResourceRequirements, error) {
+	im, err := ds.GetInstanceManager(imName)
+	if err != nil {
+		return nil, err
+	}
+
+	deprecatedCPUSetting, err := ds.GetSetting(types.SettingNameGuaranteedEngineCPU)
+	if err != nil {
+		return nil, err
+	}
+	if deprecatedCPUSetting.Value != "" {
+		return GetCPUResourceRequirement(deprecatedCPUSetting.Value)
+	}
+
+	lhNode, err := ds.GetNode(im.Spec.NodeID)
+	if err != nil {
+		return nil, err
+	}
+	kubeNode, err := ds.GetKubernetesNode(im.Spec.NodeID)
+	if err != nil {
+		return nil, err
+	}
+	switch im.Spec.Type {
+	case types.InstanceManagerTypeEngine:
+		if lhNode.Spec.EngineManagerCPURequest > 0 {
+			return GetCPUResourceRequirement(fmt.Sprintf("%.3f", float32(lhNode.Spec.EngineManagerCPURequest)/1000.0))
```

Comment:
instead of throwing in a float, just pass a milli string, ParseQuantity will handle that correctly.